### PR TITLE
Ransac

### PR DESCRIPTION
**Pull request by guidoAI**
Added an easy wrapper around the pprz_svd_solve_float method, called fit_linear_model, to which you can pass a 1D target array and a 2D array of n_samples x Dimension for making a linear fit.

Around this linear fit function, I also made a random sample consensus method (RANSAC), which is a robust way of doing the linear fit. The fit will then still give good results for a modest number of outliers.

### CODE REVIEW

File: sw/airborne/math/RANSAC.c [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  return sum;
+}
+
+/** Get indices without replacement.
+ *
+ * @param[in] indices_subset This will be filled with the sampled indices
+ * @param[in] n_samples The number of samples / indices.
+ * @param[in] count The function will sample n_sample numbers from the range 1, 2, 3,..., count
+ */
+
```
Comment (fvantienen) [inline sw/airborne/math/RANSAC.c]: This can optionally be removed for style ;)

**Comment by fvantienen** [comment]
I see now that the `math` folder is compiled during the tests with `gnu99`. In order to fix the continuous integration it might be best to change this to the new c++11? @gautierhattenberger

File: sw/airborne/math/RANSAC.c [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  }
+
+  // printf("Prediction = %f\n", sum);
+
+  return sum;
+}
+
+/** Get indices without replacement.
+ *
+ * @param[in] indices_subset This will be filled with the sampled indices
```
Comment (gautierhattenberger) [inline sw/airborne/math/RANSAC.c]: this param should be labeled `[out]`

File: sw/airborne/math/RANSAC.c [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  }
+
+}
+
+/** Predict the value of a sample with linear weights.
+ *
+ * @param[in] sample The sample vector of size D
+ * @param[in] weights The weight vector of size D+1
+ * @param[in] D The dimension of the sample.
+ */
```
Comment (gautierhattenberger) [inline sw/airborne/math/RANSAC.c]: add `@return` doc

**Comment by fvantienen** [comment]
`gnu99` and `c99` are not the same, so since we compile all our other code with `gnu99` my advise is to go with that.

**Comment by gautierhattenberger** [comment]
@kirkscheper yes, this fix should work

File: sw/airborne/math/RANSAC.c (hunk 1 above)
Comment (guidoAI) [reply sw/airborne/math/RANSAC.c]: I don't know what you mean with this... part of the doxygen?

File: sw/airborne/math/RANSAC.c (hunk 2 above)
Comment (guidoAI) [reply sw/airborne/math/RANSAC.c]: Done

File: sw/airborne/math/RANSAC.c (hunk 3 above)
Comment (guidoAI) [reply sw/airborne/math/RANSAC.c]: Done

**Comment by guidoAI** [comment]
Added the requested lines to the Makefile, and modified the comments according to the suggestions.

**Pull request merged**